include/RAJA/policy/cuda/reduce.hpp

```diff
@@ -407,11 +407,15 @@ RAJA_DEVICE RAJA_INLINE T block_reduce(T val, T identity)
   // reduce per warp values
   if (numThreads > policy::cuda::WARP_SIZE) {
 
-    __shared__ RAJA::detail::SoAArray<T, policy::cuda::MAX_WARPS> sd;
+    // Need to separate declaration and initialization for clang-cuda
+    __shared__ unsigned char tmpsd[sizeof(RAJA::detail::SoAArray<T, policy::cuda::MAX_WARPS>)];
+    RAJA::detail::SoAArray<T, policy::cuda::MAX_WARPS> * sd = new(tmpsd) RAJA::detail::SoAArray<T, policy::cuda::MAX_WARPS>();
```

Comment:
This will cause all threads to construct the object in shared memory. Only one thread should do this.

Reply:
Ok, I'll workaround this. The original code had the same problem though?

Reply:
The original never constructed the shared objects or destroyed them